### PR TITLE
Fixed share location button bug on the profile page

### LINKED ISSUE #8566: (multiple issues) Fix UI flow and debug dashboard location input form
We have a feature where you can add your location from the dashboard -- https://publiclab.org/dashboard (while logged in):

image

You click "Add your location" and a modal popup appears, prompting you:

image

Unfortunately, as @Tlazypanda noted, this doesn't really work smoothly, both due to possible UI design gaps (no notification to the user when tags are created) and possibly bugs. @Tlazypanda noted:

> * clicking "Save" doesn't apparently do anything, because you're just dropped back on the dashboard
> * On adding location modal, the slider works by zooming in but when we save it and check it out in the profile page it is saved without any zoom.
> * Also after adding your location it still shows add location on dashboard
> * This is invoked by adding location from dashboard but on clicking share location from profile page nothing happens.
> * Also after adding location for first time, it doesn't seem to update for a second time

Even after fixing this, we also ought to add a system test for adding profile tags - like, one of these modified for the dashboard!

https://github.com/publiclab/plots2/blob/86776c2c72a505b0db563f52d1a8165c36c6eb37/test/system/post_test.rb#L188-L211

We could add one of these assertions afterwards that checks that the profile tags are properly created: 

https://github.com/publiclab/plots2/blob/86776c2c72a505b0db563f52d1a8165c36c6eb37/test/system/tag_test.rb#L40-L51

### Intended workflow

The ideal intended workflow is:

1. See "add a location" on dashboard, click it
2. See modal and check content of one or more of the inputs
3. enter lat/lon and zoom, or use the slider
4. press `Save` and see an alert (maybe?) that says "profile location was saved. Learn more about location privacy" - linking to https://publiclab.org/location-privacy
5. confirm dashboard button now says "View nearby content
6. confirm tags were added and appear on profile page

### PR DESCRIPTION
**Pull request by Manasa2850**
Fixes #8566 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Manasa2850** [comment]
@jywarren @cesswairimu could you please check if this is fine? 
Thanks in advance!

**Before:**
before

**After:**
after

**Comment by coder645** [comment]
LGTM! :+1:

**Review by coder645 — APPROVED** [review]

**Review by jywarren — COMMENTED** [review]
Oh wow, thank you! 

Would you be interested in submitting a test for this as well? A system test could confirm when you click on this that it does the right thing: 

https://github.com/publiclab/plots2/blob/main/test/system/

**Comment by Manasa2850** [comment]
Sure @jywarren I'll do that. Thank you!

**Comment by Manasa2850** [comment]
@jywarren could you please help me with this? The build error says `could not find a button 'Show your Location'` 
Thanks in advance!

**Comment by gauravahlawat81** [comment]
Screencast-from-15-01-21-022322

@Manasa2850 Idk what's causing this behaviour, but `Share your Location` works fine for me. Also as @coder645 has mentioned in #8988 , saving the user location via dashboard was working on his system but not on mine.

**Comment by cesswairimu** [comment]
@gauravahlawat81  polite reminder to kindly refrain from using gender identifying names unless you know the preferred pronouns of the contributor so that we can keep being welcoming and respectful.....Thanks a lot for all the amazing work you are doing, highly appreciated :heart:

**Comment by gauravahlawat81** [comment]
Thanks @cesswairimu , I will keep this in mind.

**Comment by Manasa2850** [comment]
Oh @gauravahlawat81 that's surprising. I'm not sure what's causing that difference.
Let's wait for @jywarren

**Comment by jywarren** [comment]
Could there be a difference in whether the test user is logged in or has location tags already (or not?) I'll look in the tests to see if i can see anything. Thanks for digging into this!

**Comment by jywarren** [comment]
and, noting, the suggestion of clearing `UserTag`s out should help to disambiguate... https://github.com/publiclab/plots2/pull/8988#issuecomment-760935165 -- thanks @gauravahlawat81.

File: test/system/tag_test.rb [hunk 1]
```diff
@@ -51,5 +51,10 @@ def setup
     end
 
   end
-
+  
+  test 'adding a location tag to a user profile' do
+    visit "/profile/jeff"
```
Comment (jywarren) [inline test/system/tag_test.rb:56]: Hmm. So, is it possible the `jeff` user has an unexpected state? I have two debugging ideas -- cc @Manasa2850 @coder645 @gauravahlawat81 

1. Manually run `$('.blurred-location-input').click()` on the JavaScript console while on a profile page in your browser to see what happens
2. we could search for/assert the existence of other things before trying to run this code, to confirm that we are in the state we expect. For example, something like:

```ruby
assert !users(:jeff).has_power_tag('lat') && !users(:jeff).has_power_tag('lon')
```

code ref: 

https://github.com/publiclab/plots2/blob/2b2bd90290f22d5a5ff709b25b1a40e11094db7e/app/models/user.rb#L171

**Review by jywarren — CHANGES_REQUESTED** [review]

File: test/system/tag_test.rb (hunk 1 above)
Comment (gauravahlawat81) [reply test/system/tag_test.rb:56]: > Manually run `$('.blurred-location-input').click()` on the JavaScript console while on a profile page in your browser to see what happens

Please refer to this video -> [Video](https://drive.google.com/file/d/1mHlux37OTWuBRPFhzbxhhUQwiXJ6RVW9/view?usp=sharing)

>we could search for/assert the existence of other things before trying to run this code, to confirm that we are in the state we expect.

Seems like a good idea to me !

Comment (jywarren) [reply test/system/tag_test.rb:56]: Hi @Manasa2850 - can you confirm you're doing exactly the steps in the video @gauravahlawat81 has posted? And if not, what exactly was different? Then we can craft or adjust the test!

Comment (Manasa2850) [reply test/system/tag_test.rb:56]: Hi @jywarren yes I've followed the same steps. I've attached a gif depicting the before-after times too (in my first comment)
Thanks!

**Comment by Manasa2850** [comment]
Closing this PR since I realized it's a bug specific to my local environment. The bug is not seen on https://publiclab.org/
Thanks everyone!

**Pull request closed without merging**